src/booleans.js

```diff
@@ -1,62 +1,94 @@
 function negate(a) {
   // your code here
+  return !(a > 0);
 };
 
 function both(a, b) {
   // your code here
+  return a > 0 && b > 0;
 };
 
 function either(a, b) {
   // your code here
+  return a > 0 || b > 0;
 };
 
 function none(a, b) {
   // your code here
+  return !(a > 0 || b > 0);
 };
 
 function one(a, b) {
   // your code here
+  if (a > 0 && b > 0) {
+    return false;
+  }
+  return a > 0 || b > 0;
 };
+  
+
 
 function truthiness(a) {
-  // your code here
+  return Boolean(a);
 };
 
 function isEqual(a, b) {
   // your code here
+  return Boolean(a === b);
 };
 
 function isGreaterThan(a, b) {
   // your code here
+  return Boolean(a > b);
 };
 
 function isLessThanOrEqualTo(a, b) {
   // your code here
+  return Boolean(a <= b);
 };
 
 function isOdd(a) {
   // your code here
+  return Boolean(a % 2);
 };
 
 function isEven(a) {
   // your code here
+  return Boolean(a % 2 == 0);
 };
 
 function isSquare(a) {
   // your code here
+  if (a < 0) return false;
+  return Boolean(Math.sqrt(a) % 1 === 0);
 };
 
 function startsWith(char, string) {
   // your code here
+  if (string.startsWith(char)) return true;
+  return false;
 };
 
 function containsVowels(string) {
   // your code here
-};
+  const vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
+  for (let i = 0; i < string.length; i += 1) {
+    if (vowels.includes(string[i])) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
While this works, its a great example of somewhere that regex would be very useful
This array could be written in the format `/[aeiou]/`

What this example means is: match any part of a given string with any character between the square brackets. In this case it will only match these exact characters in lower case, but a small modifier of the `i` flag at the end (making it `/[aeiou]/i`) makes it case insensitive, so will match both cases of anything in the brackets.
These are used with the `.match` js function, like this:
```
someString.match(/[aeiou]/i)
```
this will return either true if it finds a match, or null if not. So for this function we could use 
```
  return string.match(/[aeiou]/i);
```
and it would return `true` if the string contained a lower or upper case vowel, and `null` if not. But since we only want a `true` or `false` value to be returned, we can use that double negation (!!) method again to convert the `null` value to false:
```
  return !!string.match(/[aeiou]/i);
```
Way more concise.